### PR TITLE
Set number of databases per page to display

### PR DESCRIPTION
**Pull request by Antonio-Maranhao**
## Overview

Allows the user to set the number of databases per page.

## Testing recommendations

- Go to all dbs
- Create about 6 dbs
- At the bottom, set 'Databases per page' to 5
- Validate only 5 dbs are listed in each page
- Click on a database, then click the "<" link besides the db name at the top left
- Validate it still shows 5 dbs per page

## GitHub issue number

#1167 

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the correct tag once a new Fauxton release is made

### CODE REVIEW

**Comment by popojargo** [comment]
@Antonio-Maranhao Maybe we should store this UI configuration to a cookie/localStorage? People might find it frustrating to change the setting each time they open up Fauxton.

File: app/addons/components/actions.js [hunk 1]
```diff
@@ -37,7 +37,9 @@ function deleteDatabase (dbId) {
     });
```
Comment (garrensmith) [inline app/addons/components/actions.js]: While you here, could you change the ajax to fetch.

File: app/addons/components/components/deletedatabasemodal.js [hunk 2]
```diff
@@ -64,7 +65,7 @@ export class DeleteDatabaseModal extends React.Component {
   onDeleteClick = (e) => {
     e.preventDefault();
 
-    Actions.deleteDatabase(this.getDatabaseName());
+    Actions.deleteDatabase(this.getDatabaseName(), this.props.onSuccess);
```
Comment (garrensmith) [inline app/addons/components/components/deletedatabasemodal.js]: I'm kind of torn between rather just returning a promise from the `Actions.deleteDatabase` and then calling `this.props.onSuccess`. What do you think?

File: app/addons/databases/actions.js [hunk 3]
```diff
@@ -110,7 +108,16 @@ export default {
     FauxtonAPI.dispatch({
       type: ActionTypes.DATABASES_SETPAGE,
       options: {
-        page: page
+        page
+      }
+    });
+  },
+
+  setLimit: function (limit) {
```
Comment (garrensmith) [inline app/addons/databases/actions.js]: tedious but could you remove the `function` and make it fat arrows.

File: app/addons/databases/components.js [hunk 4]
```diff
@@ -55,6 +77,18 @@ class DatabasesController extends React.Component {
     this.setState(this.getStoreState());
   };
 
+  reloadAfterDatabaseDeleted() {
+    let page = this.state.page;
+    if (page > 1 && this.state.dbList.length === 1) {
```
Comment (garrensmith) [inline app/addons/databases/components.js]: Nicely done.

**Pull request merged**